skfda/inference/anova/anova_oneway.py

```diff
@@ -0,0 +1,290 @@
+         weights (list of int): Weights related to each sample. Each
+            weight is expected to appear in the same position as its
+            corresponding sample in the FDataGrid object.
+         p (int, optional): p of the lp norm. Must be greater or equal
+            than 1. If p='inf' or p=np.inf it is used the L infinity metric.
+            Defaults to 2.
+
+    Returns:
+        The value of the statistic.
+
+    Raises:
+        ValueError.
+
+    Examples:
+
+        >>> from skfda.inference.anova import v_asymptotic_stat
+        >>> from skfda.representation.grid import FDataGrid
+        >>> import numpy as np
+
+        We create different trajectories to be applied in the statistic and a
+        set of weights.
+
+        >>> t = np.linspace(0, 1, 50)
+        >>> x1 = t * (1 - t) ** 5
+        >>> x2 = t ** 2 * (1 - t) ** 4
+        >>> x3 = t ** 3 * (1 - t) ** 3
+        >>> fd = FDataGrid([x1, x2, x3], sample_points=t)
+        >>> weights = [10, 20, 30]
+
+        Finally the value of the statistic is calculated:
+
+        >>> v_asymptotic_stat(fd, weights)
+        0.0018159320335885969
+
+    References:
+        [1] Antonio Cuevas, Manuel Febrero-Bande, and Ricardo Fraiman. "An
+        anova test for functional data". *Computational Statistics  Data
+        Analysis*, 47:111-112, 02 2004
+    """
+    if not isinstance(fd, FData):
+        raise ValueError("Argument type must inherit FData.")
+    if len(weights) != fd.n_samples:
+        raise ValueError("Number of weights must match number of samples.")
+    if isinstance(fd, FDataBasis):
+        raise NotImplementedError("Not implemented for FDataBasis objects.")
+
+    n = fd.n_samples
+    v = 0
+    for j in range(n):
+        for i in range(j):
+            v += norm_lp(
+                fd[i] - fd[j] * np.sqrt(weights[i] / weights[j]), p=p) ** p
+    return v
+
+
+def _anova_bootstrap(fd_grouped, n_sim, p=2, random_state=None):
+    assert len(fd_grouped) > 0
+
+    n_groups = len(fd_grouped)
+    sample_points = fd_grouped[0].sample_points
+    m = len(sample_points[0])  # Number of points in the grid
+    start, stop = fd_grouped[0].domain_range[0]
+
+    sizes = [fd.n_samples for fd in fd_grouped]  # List with sizes of each group
+
+    # Estimating covariances for each group
+    k_est = [fd.cov().data_matrix[0, ..., 0] for fd in fd_grouped]
+
+    # Simulating n_sim observations for each of the n_groups gaussian processes
+    sim = [make_gaussian_process(n_sim, n_features=m, start=start, stop=stop,
+                                 cov=k_est[i], random_state=random_state)
+           for i in range(n_groups)]
+    v_samples = np.array([])
+    for i in range(n_sim):
+        fd = FDataGrid([s.data_matrix[i, ..., 0] for s in sim])
+        v_samples = np.append(v_samples, v_asymptotic_stat(fd, sizes, p=p))
+    return v_samples
+
+
+def oneway_anova(*args, n_sim=2000, p=2, return_dist=False, random_state=None):
```

Comment:
It is the default value that Cuevas uses for his simulations on his article.

Reply:
Yes, I know. But what makes it a good default?

Reply:
In fact, it is a good default value, because the standard deviation of the p-values is "small" at that number of simulations, and the execution time is not so long (around 3 seconds in my computer, as time scalates linearly with the number of simulations, as expected).

![std_pvalue](https://user-images.githubusercontent.com/23263497/77199881-dadc0d00-6ae9-11ea-81b8-61b7292a072d.jpeg)

![std_pvalue_log](https://user-images.githubusercontent.com/23263497/77200162-4faf4700-6aea-11ea-8871-108c79721798.jpeg)

![execution_t_pvalue](https://user-images.githubusercontent.com/23263497/77203313-6bb5e700-6af0-11ea-9a52-221a2ed428ff.jpeg)

Reply:
Maybe I would explicitly say that the default value can be changed in the future. We can discuss it in the next meeting.

Reply:
Done!